numpy/core/src/multiarray/_multiarray_tests.c.src

```diff
@@ -2363,6 +2363,35 @@ run_intp_converter(PyObject* NPY_UNUSED(self), PyObject *args)
     return tup;
 }
 
+PyObject *
+test_scalar_intp_converter(PyObject *dummy, PyObject *obj)
+{
+    PyArray_Dims dummy_dims;
+    int output = PyArray_IntpConverter(obj, &dummy_dims);
+    if (output != NPY_SUCCEED) {
+        return NULL;
+    } else {
+        return PyLong_FromLong(dummy_dims.ptr[0]);
```

Comment:
`dummy_dims.ptr` needs to be freed here!  So you must call `npy_free_cache_dim` (or just free).

In case you would want to test more (I am not sure it is really necessary for `PyArray_IntpConverter`), you could also use: `PyObject *result = PyArray_IntTupleFromIntp(dummy_dims.nd, dummy_dims.ptr)` then you got all of the result, and not just the first number.

Reply:
Fixed in 3e6b91c5fd6f32364413c0a8eb783e92707e4328